core/src/main/java/org/infinispan/util/function/RemovableFunction.java

```diff
@@ -0,0 +1,14 @@
+package org.infinispan.util.function;
+
+import java.util.function.Function;
+
+/**
+ * Placeholder interface used to describe a function that can be used as is for removal.  This is useful for
+ * allowing an iterator to use removal.
+ * <p>
+ * Users shouldn't ever need to use this interface directly
```

Comment:
Could users implement it too? What are the requirements for a mapping function to be `RemovableFunction`?